hugolib/page.go

```diff
@@ -1021,6 +1021,14 @@ func (p *Page) update(f interface{}) error {
 			p.Date = fi.ModTime()
 			p.Params["date"] = p.Date
 		}
+	} else if p.Date.IsZero() && p.s.Cfg.GetBool("useFilenameDateAsFallback") {
```

Comment:
Sorry, I forgot this, as this will likely be copy-pasted from the documentation: 

* If the code to create a time format string from that regexp is not too complex, then 1 config is fine.
* else just add two config vars.